code/game/machinery/buttons.dm

```diff
@@ -149,14 +150,10 @@
 		return
 
 	use_power(5)
-	icon_state = "[skin]1"
-
+	flick("[skin]1", src)
```

Comment:
Change the icon state for the duration of animation
Much better than fucking spawn ()